akd/src/append_only_zks.rs

```diff
@@ -790,6 +787,66 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    #[allow(dead_code)]
+    async fn test_nonmembership_proof_v_small() -> Result<(), AkdError> {
+        let num_nodes = 2;
+
+        let mut insertion_set: Vec<Node<Blake3>> = vec![];
+
+        for i in 0..num_nodes {
+            let mut label_arr = [0u8; 32];
+            label_arr[31] = u8::from(i);
+            let label = NodeLabel::new(label_arr, 256u32);
+            let mut input = [0u8; 32];
+            input[31] = u8::from(i);
+            let hash = Blake3Digest::new(input);
+            let node = Node::<Blake3> { label, hash };
+            insertion_set.push(node);
+        }
+        let db = AsyncInMemoryDatabase::new();
+        let mut azks = Azks::new::<_, Blake3>(&db).await?;
+        let search_label = insertion_set[0].label;
+        azks.batch_insert_leaves::<_, Blake3>(&db, insertion_set.clone()[1..2].to_vec())
```

Comment:
Let's separate the node we are searching for from the nodes we are inserting into the tree.

```
let node_to_search = generate_random_node();
let insertion_set = generate_nodes(); // uses generate_random_node()
azks.batch_insert_leaves(insertion_set);
let proof = azks.get_non_membership_proof(node_to_search.label);
// Verify proof
```